Thunder/utils/canonical_files.py

```
@@ -0,0 +1,333 @@
+    copy_media: Callable[[Message], Awaitable[Optional[Message]]]
+) -> Tuple[Optional[Dict[str, Any]], Optional[Message], bool]:
+    file_unique_id = get_uniqid(source_message)
+    if not file_unique_id:
+        return None, None, False
+
+    async with file_ingest_lock(file_unique_id):
+        existing = await get_file_by_unique_id(file_unique_id)
+        if existing:
+            try:
+                is_valid = await _is_canonical_record_valid(existing, file_unique_id)
+            except Exception:
```

Ruff (0.15.6)

[warning] Do not catch blind exception: `Exception`

(BLE001)